src/Raven.Server/Commercial/InterlockedLock.cs

```diff
@@ -0,0 +1,24 @@
+using System.Threading;
+
+namespace Raven.Server.Commercial
+{
+    public class InterlockedLock
```

Comment:
Why not use a SemaphoreSlim, which can handle this? We don't want to devise our own sync primitives is we can help it.